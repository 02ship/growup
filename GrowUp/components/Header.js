import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const Header = ({title}) => {
  return (
    <View style={styles.header}>
      <Text style={styles.text}>{title}</Text>
    </View>
  );
};

Header.defaultProps = {
  title: 'GrowUp!',
};

const styles = StyleSheet.create({
  header: {
    height: 60,
    padding: 15,
    top: 15,
    backgroundColor: '#0e9307',
    borderRadius: 100,
  },
  text: {
    color: "white",
    fontSize: 23,
    textAlign: 'center',
    fontFamily: 'Chalkduster'
  }
});

export default Header;
