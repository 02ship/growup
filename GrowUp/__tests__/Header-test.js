/**
 * @format
 */

import 'react-native';
import React from 'react';
import Header from '../components/Header';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<Header />);
});

it('should match snapshot', () => {
  const result = renderer.create(<Header />).toJSON();
  expect(result).toMatchSnapshot();
});

it('renders header', () => {
  renderer.create(<Header />);
  expect(Header.defaultProps.title).toBe("GrowUp!");
});
