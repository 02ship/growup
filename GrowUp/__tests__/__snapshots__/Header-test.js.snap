// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "height": 60,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 23,
        "textAlign": "center",
      }
    }
  >
    GrowUp!
  </Text>
</View>
`;
